pkg/scyllaclient/client_rclone.go

```diff
@@ -460,22 +462,43 @@ func (c *Client) RcloneListDirIter(ctx context.Context, host, remotePath string,
 		}
 	}
 
-	var v RcloneListDirItem
-	for dec.More() {
-		// Detect context cancellation
-		if ctx.Err() != nil {
-			return ctx.Err()
+	errCh := make(chan error)
+	go func(ch chan error) {
+		var v RcloneListDirItem
+		for dec.More() {
+			// Detect context cancellation
+			if ctx.Err() != nil {
+				ch <- ctx.Err()
+				break
+			}
+
```

Comment:
Could we get rid of that?

Reply:
Yeah shouldn't be needed, if it's cancelled or we time out `More()` should come back false.